packages/translated-components/src/index.js

```diff
@@ -61,17 +62,33 @@ const moneyFormat = (language) => ({
 
 const preHeat = (translations, format) => (
   reduce(translations, (acc, o, language) => {
-    acc[language] = reduce(o, (acc, v, k) => {
-      acc[k] = new IntlFormat(v, kebabCase(language), {...moneyFormat(language), ...format})
-      return acc
-    }, {})
+    acc[language] = preheatValues(o, format, language)
     return acc
   }, {})
 )
 
-const templateReducer = (values) => (acc, v, k) => {
-  acc[k] = v.format(values)
-  return acc
+const preheatValues = (obj, format, language) => {
+  return reduce(obj, (acc, value, key) => {
+    if (typeof value === 'object') {
+      acc[key] = preheatValues(value, format, language)
+    } else {
+      acc[key] = new IntlFormat(value, kebabCase(language), {...moneyFormat(language), ...format})
+    }
+    return acc
+  }, {})
+}
+
+const templateReducer = (values) => partial(templateIteratee, values)
```

Comment:
It's being past to `lodash.reduce`. If you don't partially apply the additional piece of data I need to hold in closure, then it will never have the `values` key needed for populating the templated localisation string.

The alternative would just be to do the second level of reduction in-line, but it means you can only handle one level deep, and looks like duplicated code...but if it'd be clearer to read...